engine/access/state_stream/backend/handler_test.go

```diff
@@ -508,6 +511,75 @@ func TestEventStream(t *testing.T) {
 	}
 }
 
+// TestGetRegisterValues tests the register values.
+func TestGetRegisterValues(t *testing.T) {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	testHeight := uint64(1)
+
+	// load test register IDs + values
+	testIds := flow.RegisterIDs{
+		flow.UUIDRegisterID(0),
+		flow.AccountStatusRegisterID(unittest.AddressFixture()),
+		unittest.RegisterIDFixture(),
+	}
+	testValues := []flow.RegisterValue{
+		[]byte("uno"),
+		[]byte("dos"),
+		[]byte("tres"),
+	}
+	invalidIDs := append(testIds, flow.RegisterID{}) // valid + invalid IDs
+
+	api := ssmock.NewAPI(t)
+	api.On("GetRegisterValues", testIds, testHeight).Return(testValues, nil)
+	// wrong height
+	api.On("GetRegisterValues", testIds, testHeight+1).Return(nil, storage.ErrHeightNotIndexed)
+	// invalid register
+	api.On("GetRegisterValues", invalidIDs, testHeight).Return(nil, storage.ErrNotFound)
```

Comment:
I think it would be more clear to pair the mocks response setup with the call. I would even break it into multiple tests.